src/ndarray.jl

```diff
@@ -681,25 +681,42 @@ function div_from!(dst :: NDArray, arg :: Union{Real, NDArray})
   end
 end
 
+"""
+Elementwise division of NDArray
+"""
+div(x::NDArray, y::NDArray) = _div(x, y)
+div(x::NDArray, s::Real) = _div_scalar(x, scalar=s)
+div(s::Real, x::NDArray) = _rdiv_scalar(x, scalar=s)
+
 import Base: /
 """
     ./(arg0 :: NDArray, arg :: Union{Real, NDArray})
 
 Elementwise dividing an `NDArray` by a scalar or another `NDArray` of the same shape.
 """
-@compatdot function Base.broadcast(::typeof(/), arg0 :: NDArray, arg :: Union{Real, NDArray})
-  ret = copy(arg0, context(arg0))
-  div_from!(ret, arg)
+@compatdot function Base.broadcast(::typeof(/), arg0 :: NDArray,
+                                   arg :: Union{Real, NDArray})
+  div(arg0, arg)
+end
+
+@compatdot function Base.broadcast(::typeof(/), arg0 :: Real, arg :: NDArray)
+  div(arg0, arg)
 end
 
 """
     /(arg0 :: NDArray, arg :: Real)
 
-Divide an `NDArray` by a scalar. Matrix division (solving linear systems) is not implemented yet.
+Divide an `NDArray` by a scalar.
+Matrix division (solving linear systems) is not implemented yet.
 """
-function /(arg0 :: NDArray, arg :: Real)
-  arg0 ./ arg
-end
+/(arg0 :: NDArray, arg :: Real) = div(arg0, arg)
+
+"""
+    /(arg0 :: Real, arg :: NDArray)
+
+Elementwise Divide a scalar by an `NDArray`.
+"""
+/(arg0 :: Real, arg :: NDArray) = div(arg0, arg)
```

Comment:
I am not a fan of this since it it no longer Julia semantics

Reply:
hmm ... I misunderstood that julia 0.5 need it for broadcasting.
(maybe I should not write code in early morning :p

Reply:
Oh... we still need it because of Julia 0.6's broadcasting fusion

```julia
julia> expand(:(1 ./ mx.zeros(4)))
:($(Expr(:thunk, CodeInfo(:(begin 
        $(Expr(:thunk, CodeInfo(:(begin 
        global ##5#6
        const ##5#6
        $(Expr(:composite_type, Symbol("##5#6"), :((Core.svec)()), :((Core.svec)()), :(Core.Function), :((Core.svec)()), false, 0))
        return
    end))))
        $(Expr(:method, false, :((Core.svec)((Core.svec)(##5#6, Core.Any), (Core.svec)())), CodeInfo(:(begin 
        return 1 / #temp#
    end)), false))
        #5 = $(Expr(:new, Symbol("##5#6")))
        SSAValue(0) = #5
        SSAValue(1) = ((Core.getfield)(mx, :zeros))(4)
        return (Base.broadcast)(SSAValue(0), SSAValue(1))
    end)))))
```

Reply:
We need to solve #170. Before we can do this.